src/pipeline/deploy/index.ts

```diff
@@ -0,0 +1,48 @@
+import comment from '../../helpers/comment'
+import { deployFinalizedTemplate } from '../../helpers/commentTemplates'
+import { IDeployParams } from './types'
+import surge from '../../tenants/surge'
+import vercel from '../../tenants/vercel'
+
+const deploy = async ({
+	tokenList,
+	previewPath,
+	distFolder,
+	mountedUrl,
+	gitCommitSha,
+	outputUrl,
+	duration,
+	image,
+}: IDeployParams) => {
+	const { surgeDeploy } = surge()
+	const { vercelDeploy, returnVercelUrl } = vercel()
+
+	if (tokenList.surge) {
+		await surgeDeploy({
+			tokenList,
+			distFolder,
+			mountedUrl,
+		})
+	}
+
+	if (tokenList.vercel) {
+		await vercelDeploy({
+			tokenList,
+			distFolder,
+			previewPath,
+		})
+	}
```

Comment:
Salve Marcolas! Como vai?
Coisa boba, beeeem boba mesmo. Acho que no caso do surgeDeploy e do vercelDeploy, como eles são específicos a gente poderia mandar diretão uma prop `token` ao invés de descer a lista de tokens, inclusive acho que o destructuring poderia ser reaproveitados pelas condicionais. Então acredito que teríamos algo como: 

```suggestion
const { surge: surgeToken, vercel: vercelToken } = tokenList

	if (surgeToken) {
		await surgeDeploy({
			token: surgeToken,
			distFolder,
			mountedUrl,
		})
	}

	if (vercelToken) {
		await vercelDeploy({
			token: vercelToken,
			distFolder,
			previewPath,
		})
	}
```